awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
+            any([len(events) == 2000 for events in self.buff.values()]) or
+            (tz_now() - self.last_write).seconds > .1
+        )
+        if should_flush:
+            now = tz_now()
+            for cls, events in self.buff.items():
+                for e in events:
+                    e.created = e.modified = now
+                cls.objects.bulk_create(events)
+                for e in events:
```

Comment:
`bulk_create` doesn't fire signals, so we have to iterate and call `post_save` ourselves so that websocket messages are emitted.

Reply:
oof

Reply:
Notably faster than what we're doing now :)